pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -266,19 +305,25 @@ def __init__(
         self,
         coeffs,
         observables,
+        reorder_fn=_reorder_parameters,
         pulses: List["HardwarePulse"] = None,
         settings: Union["RydbergSettings", "TransmonSettings"] = None,
     ):
         self.settings = settings
         self.pulses = [] if pulses is None else pulses
+        self.reorder_fn = reorder_fn
         super().__init__(coeffs, observables)
 
     def __call__(self, params, t):
-        params = _reorder_parameters(params, self.coeffs_parametrized)
+        params = self.reorder_fn(params, self.coeffs_parametrized)
         return super().__call__(params, t)
 
     def __add__(self, other):
         if isinstance(other, HardwareHamiltonian):
+            if not self.reorder_fn == other.reorder_fn:
+                raise ValueError(
+                    f"Cannot add two HardwareHamiltonians with different reorder functions. Received reorder_fns {self.reorder_fn} and {other.reorder_fn}. Most likely, this is because you are trying to combine hardware compatible Hamiltonians for different target systems."
```

Comment:
I think it's generally better to avoid referring directly to the user.

```suggestion
                    f"Cannot add two HardwareHamiltonians with different reorder functions. Received reorder_fns {self.reorder_fn} and {other.reorder_fn}. This is likely due to an attempt to add hardware compatible Hamiltonians for different target systems."
```

Reply:
I think it's generally convention in professional writing 🤔